feat: add new NSE config

danielvladco opened the pull request.

danielvladco committed:
feat: add new NSE config

tiswanso commented:
in the PR description could you add some details on what the changeset involved?  Definitely there's a lot that changed due to go format but there's a change to the organization of the NSE as well so I'm trying to follow that a bit and wondering why it was required.  IMO, you might be treading pretty heavily due to a desire to refactor a lot and I don't feel that's a big priority in the initial PR for this.  But maybe as I understand the changeset better I'll see why.

tiswanso commented on examples/vl3_basic/helm/vl3/values.yaml:
```diff
@@ -10,7 +10,7 @@ pullPolicy: Always
   nsr:
     port: 5005
-    # addr: 
+    addr: localhost
```
don't change this...  `localhost:5005` means something different than `:5005`

danielvladco replied:
yes I know this, I did this change because we have this condition `{{- if .Values.cnns.nsr.addr }}` in  `vl3-nse-ucnf.tpl` so I used some random value to trigger the rendering of the new annotations

tiswanso commented on examples/vl3_basic/helm/vl3/templates/vl3-nse-ucnf.tpl:
```diff
@@ -85,12 +85,12 @@ data:
-        prefixpool: {{ .Values.ipam.prefixPool | quote }}
-        routes: []
-      ifname: "endpoint0"
+        app: vl3-nse
```
don't change this... we need to append serviceName to the app label

danielvladco replied:
fixed

tiswanso commented on examples/vl3_basic/helm/vl3/templates/vl3-nse-ucnf.tpl:
```diff
@@ -85,12 +85,12 @@ data:
+        app: vl3-nse
+      cnns:
+        name: {{ .Values.nsm.serviceName | quote }}
+        address: "{{ .Values.cnns.nsr.addr }}:{{ .Values.cnns.nsr.port }}"
```
the NSE discovery component needs the labels registered with the NSR to be `cnns/nsr.addr` and `cnns/nsr.port`

danielvladco replied:
added the labels in new commit

tiswanso commented on examples/vl3_basic/vl3-nse/cmd/nse.go:
```diff
@@ -127,44 +130,32 @@ func main() {
 
-	// Capture signals to cleanup before exiting
-	c := tools.NewOSSignalChannel()
+	pe := config.NewProcessEndpoints(backend, cnfConfig.Endpoints, configuration, CompositeEndpointPlugin)
```
did you have a need to remove the common UcnfNse layer?  The intent of that was to make it so a specific NSE instance could only add its additional composite endpoints to the common UCNF composite endpoints and make this `NewProcessEndpoints` and `pe.Process` done in the common UCNF code.  The idea being that a new type of vppagent NSE would be simpler to implement.

danielvladco replied:
I removed It because the new config doesn't have InitActions and without the init actions there was so little code in the `UcnfNse` implementation that it seemed like a useless layer to me
as you said it just wraps `config.NewProcessEndpoints` and `pe.Process` in a new object ...

danielvladco replied:
Added back UcnfNse for compatibility with the old code and for less confusion

danielvladco commented:
@tiswanso , I did go fmt, because to my usual habits of always formatting... (agree shouldn't have done it right now)

In the structure of the project the only change was that I got rid of ucnf package and put it's contents in main. Also I removed Actions and Init actions because we don't have them in the new config.

Other than that I just replaced `Endpoint` with `nseconfig.Endpoint`

I also made sure it works as it used to work before.

danielvladco committed:
fix: correct k8s manifests namings labels and addresses

danielvladco committed:
refactor: re-introduce ucnf
